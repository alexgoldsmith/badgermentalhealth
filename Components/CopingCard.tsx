import React, {useState} from 'react';
import {Button, Text, View, TouchableOpacity, TextInput} from 'react-native';
import getStyles from '../Styling/Styling'

type cCard = {
	emotion: string,
	text: string
}

type cardProps = {
	ccard: cCard,
	editCard: (cCard: cCard) => void,
	delCard: () => void
}

const styles = getStyles();

const CopingCard: React.FC<cardProps> = ({ccard, editCard, delCard}) => {
	const [isEditing, setEditing] = useState<boolean>();
	const [emotion, setEmotion] = useState<string>(ccard.emotion);
	const [text, setText] = useState<string>(ccard.text);

	return (
		<View style={[styles.card, {width: 250}]}>
			<TouchableOpacity 
				style={{display: 'flex', alignItems: 'center', justifyContent: 'center', width: 200}}
				onPress={() => setEditing(!isEditing)}
			>
				<Text style={{fontSize: 14, fontWeight: 'bold'}}>{ccard.emotion}</Text>
				<Text style={{fontSize: 14}}>{ccard.text}</Text>
			</TouchableOpacity>
			{isEditing ?
			<>
				<TextInput
					onChangeText={setEmotion}
					value={emotion}
					secureTextEntry={false}
					placeholder="Emotion"
					style={{backgroundColor: 'white', height: 40, width: 150, padding: 10, marginTop: 10, marginBottom: 10, borderColor: '#000', borderWidth: 1, borderRadius: 5}}
				/>
				<TextInput
					onChangeText={setText}
					value={text}
					secureTextEntry={false}
					placeholder="Description"
					style={{backgroundColor: 'white', height: 40, width: 150, padding: 10, marginTop: 10, marginBottom: 10, borderColor: '#000', borderWidth: 1, borderRadius: 5}}
				/>
				<View style={{display: 'flex', flexDirection: 'row'}}>
					<Button
						onPress={() => {editCard({emotion: emotion, text: text}); setEditing(false)}}
						title='Save Changes'
						color='green'
					/>
					<Button
						onPress={() => delCard()}
						title='Delete Card'
						color='red'
					/>
				</View>
			</>: <></>
			}
        </View>
	)
}

export default CopingCard;