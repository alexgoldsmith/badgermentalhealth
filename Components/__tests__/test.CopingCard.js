// https://jestjs.io/docs/snapshot-testing#snapshot-testing-with-jest
// components/__tests__/Hello.tsx
import React from 'react';
import renderer from 'react-test-renderer';

import CopingCard from '../CopingCard';

it('renders correctly with defaults', () => {
  const component = renderer
    .create(<CopingCard/>)
    .toJSON();
  expect(component).toMatchSnapshot();
});