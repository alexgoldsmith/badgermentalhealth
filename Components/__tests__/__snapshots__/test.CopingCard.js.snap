// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "display": "flex",
        "justifyContent": "center",
        "margin": 20,
        "padding": 20,
      },
      Object {
        "width": 250,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      Test emotion
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      Test text
    </Text>
  </View>
</View>
`;
